File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    const temp = new ListNode(0);
    temp.next = head
    console.log(temp, head)

    let slow = temp;
    let fast = temp;

    for (let i = 0; i< n; i++){
        fast = fast.next
    }

    while(fast.next !== null){
        fast = fast.next
        slow = slow.next
    }

    slow.next = slow.next.next
    return temp.next
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        nodes = []
        temp = ListNode(None, head)
        node = temp

        while node:
            nodes.append(node)
            node = node.next
        
        nodes[-n - 1].next = nodes[-n - 1].next.next

        return temp.next